src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs

```diff
@@ -305,22 +311,37 @@ private async Task ListenForMessages(CancellationToken cancellationToken)
 
             if (handlerToAwait != null)
             {
-                try
+                if (this.ParallelMessageProcessing)
                 {
-                    await handlerToAwait;
+                    _ = Task.Run(() =>
```

Comment:
So are you still purposely using Task.Run to kick it off to a new thread? I suppose that makes sense, but please add a comment indicating that since right now it seems a bit odd to use Task.Run when you aren't awaiting the result.

Reply:
added comment